ethereumj-core/src/main/java/org/ethereum/sync/FastSyncManager.java

```diff
@@ -409,11 +412,14 @@ public void onSuccess(List<Pair<byte[], byte[]>> result) {
                         try {
                             synchronized (FastSyncManager.this) {
                                 logger.trace("Received " + result.size() + " nodes (of " + hashes.size() + ") from peer: " + idle);
+                                idle.getNodeStatistics().eth63NodesRequested.add(hashes.size());
+                                idle.getNodeStatistics().eth63NodesRetrieveTime.add(System.currentTimeMillis() - reqTime);
                                 for (Pair<byte[], byte[]> pair : result) {
                                     TrieNodeRequest request = pendingNodes.get(pair.getKey());
                                     if (request == null) {
                                         long t = System.currentTimeMillis();
                                         logger.debug("Received node which was not requested: " + toHexString(pair.getKey()) + " from " + idle);
+                                        idle.disconnect(ReasonCode.USELESS_PEER);
```

Comment:
This case could not be related to the meaning of log message standing above. It may be a result of  `pendingNodes` purging due to `REQUEST_TIMEOUT`. Isn't it too rude to disconnect peer with `USELESS_PEER` reason in that case? Btw, log message could also be fixed to avoid misunderstanding in the future.